File: backend/src/main/java/com/newwork/employee/service/AuthService.java
package com.newwork.employee.service;

import com.newwork.employee.dto.request.LoginRequest;
import com.newwork.employee.dto.response.AuthResponse;

/**
 * Service contract for authentication operations.
 */
public interface AuthService {

    /**
     * Authenticate user with email and password
     *
     * @param loginRequest containing email and password
     * @return AuthResponse with JWT token and user details
     */
    AuthResponse login(LoginRequest loginRequest);

    /**
     * Switch to a different user (demo feature for testing)
     *
     * @param email of the user to switch to
     * @return AuthResponse with new JWT token and user details
     */
    AuthResponse switchUser(String email);
}

File: frontend/src/features/auth/api/auth-service.ts
import { httpClient } from '../../../lib/http-client';
import type { AuthResponse, LoginRequest, SwitchUserRequest } from '../types';

export const authService = {
  /**
   * Authenticate user with email and password
   */
  login: async (credentials: LoginRequest): Promise<AuthResponse> => {
    const response = await httpClient.post<AuthResponse>('/api/auth/login', credentials);
    return response.data;
  },

  /**
   * Switch to a different user (demo feature for testing roles)
   */
  switchUser: async (request: SwitchUserRequest): Promise<AuthResponse> => {
    const response = await httpClient.post<AuthResponse>('/api/auth/switch-user', request);
    return response.data;
  },
};
